enterprise/internal/authz/github/github.go

```diff
@@ -254,13 +256,15 @@ func (p *Provider) fetchUserPermsByToken(ctx context.Context, accountID extsvc.A
 			}
 		}
 
-		// If a valid cached value was found, use it and continue
-		if len(group.Repositories) > 0 {
+		// If a valid cached value was found, use it and continue. Check for a nil,
+		// because it is possible this cached group does not have any repositories, in
+		// which case it should have a non-nil length 0 slice of repositories.
+		if group.Repositories == nil {
```

Comment:
This change is what's breaking the test. I'll figure out how to do that in a way that preserves legacy behavior.

Reply:
OH! Confirmed that I write bad code. This should be:

```suggestion
		if group.Repositories != nil {
```

we want _non nil_! I even wrote it in the docstring 😆

Reply:
@mollylogue looks like this is green now!

Reply:
💯